internal/loader/configuration_setting_loader.go

```diff
@@ -679,3 +778,108 @@ func createSecretClients(ctx context.Context, acp acpv1.AzureAppConfigurationPro
 
 	return secretClients, nil
 }
+
+func parseSecretType(secretType string) (corev1.SecretType, error) {
+	secretTypeMap := map[string]corev1.SecretType{
+		"opaque":            corev1.SecretTypeOpaque,
+		"kubernetes.io/tls": corev1.SecretTypeTLS,
+	}
+
+	if parsedType, ok := secretTypeMap[secretType]; ok {
+		if parsedType != corev1.SecretTypeTLS {
+			return "", fmt.Errorf("secret type %q is not supported", secretType)
```

Comment:
So we don't allow customers to specify secretType `opaque` via the preserved tag?

Reply:
The default Secret being generated is `opaque`. There's no reasonable circumstance to specify `opaque` as preserved tag